blockscout-ens/bens-logic/src/subgraphs_reader/sql/transaction_history.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{
+    entity::subgraph::domain_event::DomainEventTransaction, subgraphs_reader::SubgraphReadError,
+};
+use lazy_static::lazy_static;
+use sqlx::postgres::PgPool;
+use tera::{Context, Tera};
+
+pub async fn find_transaction_events(
+    pool: &PgPool,
+    schema: &str,
+    id: &str,
+) -> Result<Vec<DomainEventTransaction>, SubgraphReadError> {
+    let sql =
+        sql_events_of_domain(schema).map_err(|e| SubgraphReadError::Internal(e.to_string()))?;
+    let transactions: Vec<DomainEventTransaction> =
+        sqlx::query_as(&sql).bind(id).fetch_all(pool).await?;
+    Ok(transactions)
+}
+
+const SQL_HISTORY_TEMPLATE: &str = include_str!("history.sql");
+
+lazy_static! {
+    pub static ref TEMPLATES: Tera = {
+        let mut tera = Tera::default();
+        tera.add_raw_template("history.sql", SQL_HISTORY_TEMPLATE)
+            .expect("failed to parse history.sql");
+        tera.autoescape_on(vec![".sql"]);
+        tera
+    };
+    pub static ref DEFAULT_HISTORY_CONTEXT: Context = {
+        Context::from_value(serde_json::json!({
+            "domain_event_tables": [
+                "transfer",
+                "new_owner",
+                "new_resolver",
+                "new_ttl",
+                "wrapped_transfer",
+                "name_wrapped",
+                "name_unwrapped",
+                "fuses_set",
+                "expiry_extended",
+            ],
+            "resolver_event_tables": [
+                "addr_changed",
+                "multicoin_addr_changed",
+                "name_changed",
+                "abi_changed",
+                "pubkey_changed",
+                "text_changed",
+                "contenthash_changed",
+                "interface_changed",
+                "authorisation_changed",
+                "version_changed",
+            ],
+            "registration_event_tables": [
+                "name_registered",
+                "name_renewed",
+                "name_transferred"
+            ]
+        }))
+        .expect("failed to load history context")
+    };
+}
```

Comment:
Wouldn't those values be different depending on the chain we are processing?

Reply:
For now, we have only 2 subgraphs, and their schema is identical, so I decided to use common values for them

Reply:
If we will face problem of different schemas for events, we will change this code, it won't be hard